[EWS] buildbot master unexpectedly stopped and builds failed with error - Master Shutdown

aj062 opened the pull request:
#### b71f2fd0d5382e0b1e4e12e8d522632c11a397d6
<pre>
[EWS] buildbot master unexpectedly stopped and builds failed with error - Master Shutdown
<a href="https://bugs.webkit.org/show_bug.cgi?id=303324">https://bugs.webkit.org/show_bug.cgi?id=303324</a>
<a href="https://rdar.apple.com/163674424">rdar://163674424</a>

Reviewed by Alexey Proskuryakov.

buildbot was abruptly stopping when a JSC test run resulted in 100,000+ test failures. The reason for that was:
- These test failures generated huge list of failed tests (just the list text being over 8 MB).
- These failures were parsed and stored as build property twice (jsc_stress_test_failures and jsc_stress_test_failures_filtered)
- That caused build properties to be over 16 MB.
- After the jscore-tests step was complete, the build-master sent websocket message containing these build properties
  to crossbar (in order to sync to other master).
- This websocket message was over 16MB, which was the already very high limit we set for crossbar message size.
- crossbar saw WebSocket message exceeding the liit and dropped the connection to buildbot
- buildbot saw that the connection to crossbar was dropped, didn&apos;t know how to recover, and shut itself down, causing
  all the builds to stop.

Fixing the issue by preventing these build properties from being huge.

* Tools/CISupport/ews-build/steps.py:
(RunJavaScriptCoreTests):
(RunJavaScriptCoreTests.run):

Canonical link: <a href="https://commits.webkit.org/303694@main">https://commits.webkit.org/303694@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/a7d8cae90a0102928336260215d4ae2f6d54486b

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows | Apple Internal |
| ----- | ---------------------- | ------- |  ----- |  --------- | ------ |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/133385 "Passed style check") | [  ~~🛠 ios~~](https://ews-build.webkit.org/#/builders/159/builds/5886 "EWS skipped this build as PR had skip-ews label when EWS attempted to process it.") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/44521 "Built successfully") | [  ~~🛠 wpe~~](https://ews-build.webkit.org/#/builders/5/builds/140940 "EWS skipped this build as PR had skip-ews label when EWS attempted to process it.") | [  ~~🛠 win~~](https://ews-build.webkit.org/#/builders/59/builds/85432 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | ⏳ 🛠 ios-apple 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/135255 "Passed tests") | [  ~~🛠 ios-sim~~](https://ews-build.webkit.org/#/builders/155/builds/6399 "EWS skipped this build as PR had skip-ews label when EWS attempted to process it.") | [  ~~🛠 mac-AS-debug~~](https://ews-build.webkit.org/#/builders/156/builds/5751 "EWS skipped this build as PR had skip-ews label when EWS attempted to process it.") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/5/builds/140940 "EWS skipped this build as PR had skip-ews label when EWS attempted to process it.") | [  ~~🧪 win-tests~~](https://ews-build.webkit.org/#/builders/59/builds/85432 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | ⏳ 🛠 mac-apple 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/136332 "Passed tests") | [  ~~🧪 ios-wk2~~](https://ews-build.webkit.org/#/builders/155/builds/6399 "EWS skipped this build as PR had skip-ews label when EWS attempted to process it.") | [  ~~🧪 api-mac~~](https://ews-build.webkit.org/#/builders/18/builds/119563 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 api-wpe~~](https://ews-build.webkit.org/#/builders/5/builds/140940 "EWS skipped this build as PR had skip-ews label when EWS attempted to process it.") | | ⏳ 🛠 vision-apple 
| | [  ~~🧪 ios-wk2-wpt~~](https://ews-build.webkit.org/#/builders/155/builds/6399 "EWS skipped this build as PR had skip-ews label when EWS attempted to process it.") | [⏳ 🧪 api-mac-debug](https://ews-build.webkit.org/#/builders/macOS-Tahoe-Debug-API-Tests-EWS "Waiting to run tests") | [⏳ 🛠 wpe-cairo ](https://ews-build.webkit.org/#/builders/WPE-Cairo-Build-EWS "Waiting in queue, processing has not started yet") | | 
| | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/155/builds/6399 "EWS skipped this build as PR had skip-ews label when EWS attempted to process it.") | [  ~~🧪 mac-wk1~~](https://ews-build.webkit.org/#/builders/135/builds/37672 "EWS skipped this build as PR had skip-ews label when EWS attempted to process it.") | [  ~~🛠 gtk~~](https://ews-build.webkit.org/#/builders/2/builds/143587 "EWS skipped this build as PR had skip-ews label when EWS attempted to process it.") | | 
| | [  ~~🛠 vision~~](https://ews-build.webkit.org/#/builders/153/builds/5556 "EWS skipped this build as PR had skip-ews label when EWS attempted to process it.") | [  ~~🧪 mac-wk2~~](https://ews-build.webkit.org/#/builders/136/builds/38259 "EWS skipped this build as PR had skip-ews label when EWS attempted to process it.") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/2/builds/143587 "EWS skipped this build as PR had skip-ews label when EWS attempted to process it.") | | 
| [✅ 🧪 services](https://ews-build.webkit.org/#/builders/28/builds/132813 "Passed tests") | [  ~~🛠 vision-sim~~](https://ews-build.webkit.org/#/builders/160/builds/5638 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/156/builds/5751 "EWS skipped this build as PR had skip-ews label when EWS attempted to process it.") | [  ~~🧪 api-gtk~~](https://ews-build.webkit.org/#/builders/2/builds/143587 "EWS skipped this build as PR had skip-ews label when EWS attempted to process it.") | | 
| | [⏳ 🧪 vision-wk2 ](https://ews-build.webkit.org/#/builders/visionOS-26-Simulator-WK2-Tests-EWS "Waiting in queue, processing has not started yet") | [  ~~🧪 mac-wk2-stress~~](https://ews-build.webkit.org/#/builders/8/builds/115820 "EWS skipped this build as PR had skip-ews label when EWS attempted to process it.") | [  ~~🛠 playstation~~](https://ews-build.webkit.org/#/builders/134/builds/59306 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/20627 "Built successfully and passed tests") | [  ~~🛠 tv~~](https://ews-build.webkit.org/#/builders/158/builds/5611 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-intel-wk2~~](https://ews-build.webkit.org/#/builders/137/builds/34151 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | | 
| | [  ~~🛠 tv-sim~~](https://ews-build.webkit.org/#/builders/157/builds/5457 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | | | 
| | [  ~~🛠 watch~~](https://ews-build.webkit.org/#/builders/163/builds/5700 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | | | 
| | [  ~~🛠 watch-sim~~](https://ews-build.webkit.org/#/builders/152/builds/5567 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | | | 
<!--EWS-Status-Bubble-End-->

aproskuryakov reviewed (approved):
Of course, people sometimes want to see all test failures for bigger changes and mass rebaselines. Avoiding buildbot shutdown is better than shutting down, but not perfect.

Perhaps we could at least have a larger threshold than just 60? A low threshold for layout tests is also meant to stop them from running quickly, but there doesn't seem to be any need to trim it so much after the fact.

aj062 committed:
[EWS] buildbot master unexpectedly stopped and builds failed with error - Master Shutdown
https://bugs.webkit.org/show_bug.cgi?id=303324
rdar://163674424

Reviewed by Alexey Proskuryakov.

buildbot was abruptly stopping when a JSC test run resulted in 100,000+ test failures. The reason for that was:
- These test failures generated huge list of failed tests (just the list text being over 8 MB).
- These failures were parsed and stored as build property twice (jsc_stress_test_failures and jsc_stress_test_failures_filtered)
- That caused build properties to be over 16 MB.
- After the jscore-tests step was complete, the build-master sent websocket message containing these build properties
  to crossbar (in order to sync to other master).
- This websocket message was over 16MB, which was the already very high limit we set for crossbar message size.
- crossbar saw WebSocket message exceeding the liit and dropped the connection to buildbot
- buildbot saw that the connection to crossbar was dropped, didn't know how to recover, and shut itself down, causing
  all the builds to stop.

Fixing the issue by preventing these build properties from being huge.

* Tools/CISupport/ews-build/steps.py:
(RunJavaScriptCoreTests):
(RunJavaScriptCoreTests.run):

Canonical link: https://commits.webkit.org/303694@main
